src/app/codegen-data-model-provider/CodegenDataModelProvider_Write.cpp

```diff
@@ -322,7 +319,45 @@ DataModel::ActionReturnStatus CodegenDataModelProvider::WriteAttribute(const Dat
     if (!request.operationFlags.Has(DataModel::OperationFlags::kInternal))
     {
         VerifyOrReturnError(!isReadOnly, Status::UnsupportedWrite);
+    }
 
+    // ACL check for non-internal requests
+    bool checkAcl = !request.operationFlags.Has(DataModel::OperationFlags::kInternal);
+
+    // For chunking, ACL check is not re-done if the previous write was successful for the exact same
+    // path. We apply this everywhere as a shortcut, although realistically this is only for AccessControl cluster
+    if (checkAcl && request.previousSuccessPath.has_value())
+    {
+        checkAcl = ((request.path.mEndpointId != request.previousSuccessPath->mEndpointId) ||
+                    (request.path.mClusterId != request.previousSuccessPath->mClusterId) ||
+                    (request.path.mAttributeId != request.previousSuccessPath->mAttributeId));
+    }
+
+    // If not writable, do unsupported write instead of unsupported-access
```

Comment:
Is this comment still relevant? If so I am not following and it doesn't make sense to me

Reply:
You are correct - this was a left over from when the bug was we need to say "unsupported write" instead of access denied. This is no handled and commented above. Removed this.